src/java.base/share/classes/jdk/internal/util/StaticProperty.java

```diff
@@ -276,4 +308,109 @@ public static String osArch() {
+            case 0 -> USER_SCRIPT;
+            case 1 -> USER_SCRIPT_DISPLAY;
+            case 2 -> USER_SCRIPT_FORMAT;
+            default -> throw new InternalError();
+        };
+    }
+
+    /**
+     * {@return the {@code user.country} system property}
+     * <strong>{@link SecurityManager#checkPropertyAccess} is NOT checked
+     * in this method. The caller of this method should take care to ensure
+     * that the returned property is not made accessible to untrusted code.</strong>
+     *
+     * @param category locale category. 0 for the base property,
+     *                 {@code Locale.Category.ordinal() + 1} for the category
+     *                 specific property
+     */
+    public static String userCountry(int category) {
+        return switch (category) {
+            case 0 -> USER_COUNTRY;
+            case 1 -> USER_COUNTRY_DISPLAY;
+            case 2 -> USER_COUNTRY_FORMAT;
+            default -> throw new InternalError();
+        };
+    }
+
+    /**
+     * {@return the {@code user.variant} system property}
+     * <strong>{@link SecurityManager#checkPropertyAccess} is NOT checked
+     * in this method. The caller of this method should take care to ensure
+     * that the returned property is not made accessible to untrusted code.</strong>
+     *
+     * @param category locale category. 0 for the base property,
+     *                 {@code Locale.Category.ordinal() + 1} for the category
+     *                 specific property
+     */
+    public static String userVariant(int category) {
+        return switch (category) {
+            case 0 -> USER_VARIANT;
+            case 1 -> USER_VARIANT_DISPLAY;
+            case 2 -> USER_VARIANT_FORMAT;
+            default -> throw new InternalError();
+        };
+    }
+
+    /**
+     * {@return the {@code user.extensions} system property}
+     * <strong>{@link SecurityManager#checkPropertyAccess} is NOT checked
+     * in this method. The caller of this method should take care to ensure
+     * that the returned property is not made accessible to untrusted code.</strong>
+     *
+     * @param category locale category. 0 for the base property,
+     *                 {@code Locale.Category.ordinal() + 1} for the category
+     *                 specific property
+     */
+    public static String userExtensions(int category) {
+        return switch (category) {
+            case 0 -> USER_EXTENSIONS;
+            case 1 -> USER_EXTENSIONS_DISPLAY;
+            case 2 -> USER_EXTENSIONS_FORMAT;
+            default -> throw new InternalError();
+        };
+    }
+
+    /**
+     * {@return the {@code user.region} system property}
+     * <strong>{@link SecurityManager#checkPropertyAccess} is NOT checked
+     * in this method. The caller of this method should take care to ensure
+     * that the returned property is not made accessible to untrusted code.</strong>
+     */
+    public static String userRegion() {
+        return USER_REGION;
+    }
```

Comment:
Using methods to retrieve these makes is more complicated.
The bleeding of the enum values outside of Locale is undesirable.
Since the property values are final strings, I suggest just making the fields public and keep the mapping local to the Locale class.

Reply:
As Alan commented, I will not use `Locale.Category.ordinal()` but instead use the properties keys. Would that address your suggestion?

Reply:
Ended up replacing public methods to public fields